project/apps/api-gateway/src/questions/questions.controller.ts

```diff
@@ -48,10 +58,14 @@ export class QuestionsController {
   @Put(':id')
   async updateQuestion(
     @Param('id') id: string,
-    @Body() updateQuestionDto: UpdateQuestionDto,
+    @Body(new ZodValidationPipe(updateQuestionSchema)) // validation on the body only
```

Comment:
if i have it in `UsePipes`, it will validate the param and body separate, causing validation to fail. (since param `id` will not pass the updateQuestionSchema validation check). 

btw there's an error here, `@Body(...) ...)` should only appear once.